doc/architecture/frontend.md

```diff
@@ -0,0 +1,87 @@
+# Frontend Architecture
+
+## Structure
+
+- `__mocks__/`: Contains mocked modules for tests.
+- `assets/`: Contains small asset files like SVGs.
+- `bindings/`: Contains the generated API bindings.
```

Comment:
Idk if bindings is the right word. Maybe dtos, shared_types or api_types would be more fitting

Reply:
Yes, I think I will go with `api_types` because it is unambiguous, and is easy to understand.

Reply:
I don't know `api_types` and would expect something like different types of APIs, like a functional one, one with locks, one lock-free etc. But this is probably not what should be there? If `api_types` is well-known in the React world I am okay with it, though. Please give a short explanation, though.